oximeter/instruments/src/kstat/sampler.rs

```diff
@@ -93,16 +97,31 @@ mod self_stats {
 /// An identifier for a single tracked kstat target.
 ///
 /// This opaque identifier can be used to unregister targets from the sampler.
-/// If not removed, data from the targets will be produced forever.
-#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+/// If not removed, data from the targets will be produced according to the
+/// [`ExpirationBehavior`] configured for the target.
+#[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct TargetId(u64);
 
-/// How to handle a kstat which can no longer be collected from.
+impl fmt::Debug for TargetId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl fmt::Display for TargetId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+/// When to expire kstat which can no longer be collected from.
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub enum ExpirationBehavior {
     /// Never stop attempting to produce data from this target.
     Never,
     /// Expire after a number of sequential failed collections.
+    ///
+    /// Use `Attempts(0)` to expire after the first failure.
```

Comment:
Turbo nit: I might reword to "If the payload is 0, expire after the first failure" or similar, since `Attempts(1)` also has this behavior.

Reply:
Done in 6740d1b955caa7ecb9336b1bcbc365f72a0a893d